Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLevelBuilder.h

```diff
@@ -28,13 +27,16 @@ class CmsTrackerLevelBuilderHelper {
  * Abstract Class to construct a Level in the hierarchy
  */
 template <class FilteredView>
-class CmsTrackerLevelBuilder : public CmsTrackerAbstractConstruction<FilteredView> {
+class CmsTrackerLevelBuilder {
 public:
-  void build(FilteredView&, GeometricDet*, const std::string&) override;
-  ~CmsTrackerLevelBuilder() override {}
+  virtual void build(FilteredView&, GeometricDet*, const std::string&);
+  virtual ~CmsTrackerLevelBuilder() = default;
 
 private:
+  static bool skipFirstChild;
   virtual void buildComponent(FilteredView&, GeometricDet*, const std::string&) = 0;
+  void buildLoop(DDFilteredView& fv, GeometricDet* tracker, const std::string& attribute);
+  void buildLoop(cms::DDFilteredView& fv, GeometricDet* tracker, const std::string& attribute);
```

Comment:
These two specialized functions allow us to isolate the work on the loop that modifies `fv`